src/hooks/hooks-middleware.ts

```diff
@@ -37,9 +37,13 @@ export const hooksMiddleware: Middleware<DBCore>  = {
           const {deleting, creating, updating} = dxTrans.table(tableName).hook;
           switch (req.type) {
             case 'add':
+              // Ensure that the req has keys
+              req.keys = getEffectiveKeys(downTable.schema.primaryKey, req);
```

Comment:
This would slow down all requests even when there are no hooks registered.
Not all hooks would need to have the keys so even if there was an 'adding' hook registered, it could hurt performance for hooks that wouldn't want this.